src/systems/systems/src/hydraulic/aerodynamic_model.rs

```diff
@@ -0,0 +1,669 @@
+use nalgebra::{Rotation3, Vector3};
+
+use uom::si::{
+    area::square_meter, f64::*, force::newton, length::meter,
+    mass_density::kilogram_per_cubic_meter,
+};
+
+use crate::{shared::interpolation, simulation::UpdateContext};
+
+/// Projects a 3D box with 3D size along a projection vector.
+/// Outputs the projected area on the plane normal to projection vector
+fn area_projected(size: Vector3<Length>, projection_vector: Vector3<f64>) -> Area {
+    let norm_projection_vector = projection_vector.normalize();
+
+    Area::new::<square_meter>(
+        size[0].get::<meter>() * size[2].get::<meter>() * norm_projection_vector[1]
+            + size[0].get::<meter>() * size[1].get::<meter>() * norm_projection_vector[2]
+            + size[1].get::<meter>() * size[2].get::<meter>() * norm_projection_vector[0],
+    )
+}
+
+pub struct AerodynamicModel {
+    max_drag_normal: Vector3<f64>,
+    lift_axis: Vector3<f64>,
+    lift_normal: Vector3<f64>,
+
+    lift_area: Area,
+    drag_area: Area,
+
+    has_drag: bool,
+    has_lift: bool,
+}
+impl AerodynamicModel {
+    const DEFAULT_LIFT_ANGLE_MAP_DEGREES: [f64; 7] = [-30., -18., -10., 0., 10., 18., 30.];
+    const DEFAULT_LIFT_COEFF_MAP: [f64; 7] = [0., -0.5, -1.2, 0., 1.2, 0.5, 0.];
+
+    const DEFAULT_DRAG_COEFF: f64 = 1.2;
+
+    const MIN_WIND_SPEED_TO_COMPUTE_AERO_FORCES_M_S: f64 = 0.5;
+
+    pub fn new(
+        body: &impl AerodynamicBody,
+        max_drag_normal: Option<Vector3<f64>>,
+        lift_axis: Option<Vector3<f64>>,
+        lift_normal: Option<Vector3<f64>>,
+    ) -> Self {
+        let mut obj = Self {
+            max_drag_normal: max_drag_normal.unwrap_or_default(),
+            lift_axis: lift_axis.unwrap_or_default(),
+            lift_normal: lift_normal.unwrap_or_default(),
+
+            lift_area: Area::default(),
+            drag_area: Area::default(),
+            has_drag: max_drag_normal.is_some(),
+            has_lift: lift_axis.is_some(),
+        };
+        obj.initialize_areas(body.size());
+        obj
+    }
```

Comment:
**suggestion**: As all usages currently pass `Some` value, and you'll use the default if not provided, consider not having an `Option` type.

Reply:
I'd rather keep the option just for now until some more tests are done on spoilers and gears. Lift might not be needed there but I'm still unsure :p